### PR TITLE
feat: add og image

### PR DESCRIPTION
**Pull request by Rei-x**
This pull request includes several updates and improvements across different files, primarily focusing on adding descriptions to various documentation files, updating dependencies, and enhancing the Open Graph image generation. Below are the most important changes:

### Documentation Enhancements:
* Added descriptions to multiple documentation files to provide better context and improve SEO. This includes files like `index.md`, `accessibility.md`, `best-practices.mdx`, `common-mistakes.md`, and more. [[1]](diffhunk://#diff-58ff131542fe0d765eb0f51be057d7c3608b5c39c9536f4888e187211af898deR5-R12) [[2]](diffhunk://#diff-544812ca940edda007553f7c15cbcc28f19274886c007301ad7d71bab2aaaa5dR5) [[3]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dR5) [[4]](diffhunk://#diff-8556c14065afe1973263453631ff88a688031f1c4709d118597d8475724e37b4R5) [[5]](diffhunk://#diff-26b9219dbb3b0a63ffd4d24c2407a103c2b2f9e79e3d8cd90ba664b31cfc3350R8) [[6]](diffhunk://#diff-c85796196a9ea3ccaf1577e7274dc85db963008403a154c67570c6aa71177233R5) [[7]](diffhunk://#diff-ee91ac6d5e806488de94caaf0b0cd087712d34a164ade34217fa7995ad6abf9bR5) [[8]](diffhunk://#diff-26d79e096cbe90b49aca96eaccece29641cf1567bc8ee4b936adbb58d5a271cdL3-R3)

### Dependency Updates:
* Updated `astro` to version `4.16.16` and other dependencies like `tailwindcss` and `typescript` in the `package.json` file. Additionally, removed some unused dependencies such as `dotenv`, `google-auth-library`, and `googleapis`.

### Configuration Changes:
* Added a new component override for `Head` in `astro.config.mjs` to customize the `<Head>` component used across the site.

### Open Graph Image Generation:
* Introduced a new file `src/pages/og/[...slug].ts` to generate Open Graph images dynamically using `astro-og-canvas`. This includes defining the image options and customizing the generated images with titles, descriptions, and styles. ([src/pages/og/[...slug].tsR1-R45](diffhunk://#diff-5ca3c377077ccc042ec70dbfeefc665a5b368169b30001e95168329acbb8a2a5R1-R45))
* Created a new component `src/overrides/Head.astro` to render Open Graph meta tags with the generated image URLs.

### Markdown Link Formatting:
* Updated various markdown files to format links correctly by enclosing them in angle brackets. This change was applied to files like `index.md`, `whats-incoming.md`, and others. [[1]](diffhunk://#diff-58ff131542fe0d765eb0f51be057d7c3608b5c39c9536f4888e187211af898deL46-R47) [[2]](diffhunk://#diff-58ff131542fe0d765eb0f51be057d7c3608b5c39c9536f4888e187211af898deL76-R77) [[3]](diffhunk://#diff-58ff131542fe0d765eb0f51be057d7c3608b5c39c9536f4888e187211af898deL95-R96) [[4]](diffhunk://#diff-58ff131542fe0d765eb0f51be057d7c3608b5c39c9536f4888e187211af898deL187-R202)

### CODE REVIEW

**Comment by Rei-x** [comment]
przykładowe og:

File: src/pages/og/[...slug].ts
Comment (qamarq) [inline src/pages/og/[...slug].ts:1]: dobree, lubie to

**Review by qamarq — APPROVED** [review]
niczym d0dii, nic podejrzanego nie widze, dla mnie wszystko git, approve

**Comment by qamarq** [comment]
> zykładowe og:

Dobra robota!

**Pull request merged**